### PR TITLE
Exclude `packages/app/public` from eslint in create-app template

### PR DESCRIPTION
**Pull request by taras**
## Hey, I just made a Pull Request!

Follow up to https://github.com/backstage/backstage/issues/16571

I added `packages/app/.eslintignore` with `public` to the template in `@backstage/create-app` package

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by benjdlambert** [comment]
Hmm it's a little unfortunate that this catches everything called `public`. Can we lock it down a little further and be more opinionated and then if people change they can update this path too?

**Comment by taras** [comment]
@benjdlambert what about `packages/app/public`?

**Comment by benjdlambert** [comment]
@taras thinking that we make a `packages/app/.eslintignore` with `public` in it instead? Feels a little nicer that way?

**Comment by taras** [comment]
@benjdlambert good idea, done!

File: .changeset/witty-squids-fetch.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/create-app': patch
+---
+
```
Comment (benjdlambert) [inline .changeset/witty-squids-fetch.md:4]: Probably just wanna change this changeset a little bit to be more reflective of the actual change now.

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Nice, just smol nit about changeset.

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**